Docker does not support '[]' in container names

This is mainly a problem when parametrized ui_airgun-tests are run on selenium-docker containers.
Seen for `tests/foreman/ui_airgun/test_operatingsystem.py::test_positive_create[0]`.

Error was:
```
self = <docker.client.Client object at 0x7f7eabf59160>, response = <Response [500]>, explanation = None

    def _raise_for_status(self, response, explanation=None):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
            response.raise_for_status()
        except requests.exceptions.HTTPError as e:
            if e.response.status_code == 404:
                raise errors.NotFound(e, response, explanation=explanation)
>           raise errors.APIError(e, response, explanation=explanation)
E           docker.errors.APIError: 500 Server Error: Internal Server Error ("b'{"message":"Invalid container name (test_positive_create[0]_5oo), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed"}'")

../../.local/lib/python3.6/site-packages/docker/client.py:174: APIError
```

--- a/airgun/browser.py
+++ b/airgun/browser.py
@@ -447,7 +447,12 @@ class DockerBrowser(object):
                     'Could not find docker-image "%s"; please pull it' %
                     self._get_image_name(image_version)
                 )
-
+        # Grab only the test name, get rid of square brackets from parametrize
+        # and add some random chars. E.g. 'test_positive_create_0_abc'
+        container_name = '{}_{}'.format(
+            self._name.split('.')[-1].replace('[', '_').strip(']'),
+            gen_string('alphanumeric', 3)
+        )
         self.container = self._client.create_container(
             detach=True,
             environment={
@@ -457,8 +462,7 @@ class DockerBrowser(object):
             host_config=self._client.create_host_config(
                 publish_all_ports=True),
             image=self._get_image_name(image_version),
-            name=self._name.split('.')[-1] + '_{0}'.format(
-                gen_string('alphanumeric', 3)),
+            name=container_name,
             ports=[4444],
         )
         LOGGER.debug('Starting container with ID "%s"', self.container['Id'])

